### PR TITLE
Recheck for global and venv dvc version

### LINKED ISSUE #3101: Onboarding screen stuck on install DVC even after it's installed
On Windows, latest VS Code 1.74.3, Python extension is installed.

https://user-images.githubusercontent.com/3659196/212571271-3f20b263-290f-4026-8fc4-5f5ade98856f.mov

Reloading the VS Code helped.

### PR DESCRIPTION
**Pull request by sroy3**
Fix #3101 

https://user-images.githubusercontent.com/3683420/214918032-368d542c-4d66-4bc7-bd38-ceecf8bc68ae.mov

### CODE REVIEW

File: extension/src/cli/dvc/discovery.ts [hunk 1]
```diff
@@ -229,12 +239,13 @@ export const recheckGlobal = async (
     return
   }
 
-  const version = await setup.getCliVersion(cwd, true)
-  const cliCompatible = isVersionCompatible(version)
-  const isCompatible = isCliCompatible(cliCompatible)
+  let isCompatible = await checkVersion(setup, cwd, undefined)
```
Comment (mattseddon) [inline extension/src/cli/dvc/discovery.ts:242]: [N] `undefined` is optional here.

File: extension/src/test/suite/setup/util.ts [hunk 2]
```diff
@@ -78,10 +78,9 @@ const buildSetupDependencies = (
     } as unknown as InternalCommands,
     mockRoot,
     mockRunSetup: stub(Runner, 'run').resolves(undefined),
-    mockRunSetupWithGlobalRecheck: stub(
-      Runner,
-      'runWithGlobalRecheck'
-    ).resolves(undefined),
+    mockRunSetupWithGlobalRecheck: stub(Runner, 'runWithRecheck').resolves(
```
Comment (mattseddon) [inline extension/src/test/suite/setup/util.ts:81]: [C] Rename `mockRunSetupWithGlobalRecheck` to match new function name

File: extension/src/cli/dvc/discovery.ts (hunk 1 above)
Comment (mattseddon) [reply extension/src/cli/dvc/discovery.ts:242]: Can use `await checkVersion(setup, cwd)`

File: extension/src/cli/dvc/discovery.ts [hunk 3]
```diff
@@ -229,12 +239,13 @@ export const recheckGlobal = async (
...
+  if (!isCompatible) {
+    isCompatible = await checkVersion(setup, cwd, true)
```
Comment (mattseddon) [inline extension/src/cli/dvc/discovery.ts:244]: [Q] Should we put another delay in here between the global and venv checks? Run them with an offset?

**Review by mattseddon — APPROVED** [review]

File: extension/src/cli/dvc/discovery.ts (hunk 3 above)
Comment (sroy3) [reply extension/src/cli/dvc/discovery.ts:244]: I think it might just make this slower when installing.

**Pull request merged**